apps/auth0-rules/src/add-user-metadata.ts

```diff
@@ -1,10 +1,63 @@
 import type { User } from '@asap-hub/auth';
-import type { UserMetadataResponse } from '@asap-hub/model';
+import type { gp2, UserMetadataResponse, UserResponse } from '@asap-hub/model';
 import got from 'got';
 import { URL, URLSearchParams } from 'url';
 import { handleError } from './handle-error';
 import type { Rule } from './types';
 
+type Auth0UserResponse = UserMetadataResponse | gp2.UserResponse;
+
+function isUserResponse(
+  response: UserMetadataResponse | gp2.UserResponse,
+): response is UserResponse & { algoliaApiKey: string } {
+  return 'onboarded' in (response as UserMetadataResponse);
```

Comment:
the cast does not seem necessary, nor does it look right. Are you sure its needed?

Reply:
done. The cast was from the Typescript documentation.